pykokkos/lib/ufuncs.py

```diff
@@ -3009,3 +3009,307 @@ def isfinite(view):
+    if "int" in str(view.dtype):
+        # special case defined in API std
+        return view
+    out = pk.View(view.shape, dtype=view.dtype)
+    if len(view.shape) > 3:
+        raise NotImplementedError("only up to 3D views currently supported for trunc() ufunc.")
+        
+    if "double" in str(view.dtype):
+        if view.shape == ():
+            pk.parallel_for(1, trunc_impl_1d_double, view=view, out=out)
+        elif len(view.shape) == 1:
+            pk.parallel_for(view.shape[0], trunc_impl_1d_double, view=view, out=out)
+        elif len(view.shape) == 2:
+            pk.parallel_for(view.shape[0], trunc_impl_2d_double, view=view, out=out)
+        elif len(view.shape) == 3:
+            pk.parallel_for(view.shape[0], trunc_impl_3d_double, view=view, out=out)
+
+    elif "float64" in str(view.dtype):
+        if view.shape == ():
+            pk.parallel_for(1, trunc_impl_1d_double, view=view, out=out)
+        elif len(view.shape) == 1:
+            pk.parallel_for(view.shape[0], trunc_impl_1d_double, view=view, out=out)
+        elif len(view.shape) == 2:
+            pk.parallel_for(view.shape[0], trunc_impl_2d_double, view=view, out=out)
+        elif len(view.shape) == 3:
+            pk.parallel_for(view.shape[0], trunc_impl_3d_double, view=view, out=out)
+
+    elif "float" in str(view.dtype):
+        if view.shape == ():
+            pk.parallel_for(1, trunc_impl_1d_float, view=view, out=out)
+        elif len(view.shape) == 1:
+            pk.parallel_for(view.shape[0], trunc_impl_1d_float, view=view, out=out)
+        elif len(view.shape) == 2:
+            pk.parallel_for(view.shape[0], trunc_impl_2d_float, view=view, out=out)
+        elif len(view.shape) == 3:
+            pk.parallel_for(view.shape[0], trunc_impl_3d_float, view=view, out=out)
+
+    else:
+        raise NotImplementedError
+    return out
+
+@pk.workunit
+def ceil_impl_1d_double(tid: int, view: pk.View1D[pk.double], out: pk.View1D[pk.double]):
+    out[tid] = ceil(view[tid])
+
+
+@pk.workunit
+def ceil_impl_2d_double(tid: int, view: pk.View2D[pk.double], out: pk.View2D[pk.double]):
+    for i in range(view.extent(1)):
+        out[tid][i] = ceil(view[tid][i])
+
+
+@pk.workunit
+def ceil_impl_3d_double(tid: int, view: pk.View3D[pk.double], out: pk.View3D[pk.double]):
+    for i in range(view.extent(1)):
+        for j in range(view.extent(2)):
+            out[tid][i][j] = ceil(view[tid][i][j])
+
+@pk.workunit
+def ceil_impl_1d_float(tid: int, view: pk.View1D[pk.float], out: pk.View1D[pk.float]):
+    out[tid] = ceil(view[tid])
+
+
+@pk.workunit
+def ceil_impl_2d_float(tid: int, view: pk.View2D[pk.float], out: pk.View2D[pk.float]):
+    for i in range(view.extent(1)):
+        out[tid][i] = ceil(view[tid][i])
+
+
+@pk.workunit
+def ceil_impl_3d_float(tid: int, view: pk.View3D[pk.float], out: pk.View3D[pk.float]):
+    for i in range(view.extent(1)):
+        for j in range(view.extent(2)):
+            out[tid][i][j] = ceil(view[tid][i][j])
```

Comment:
Is this right? I'm afraid we may have to add some custom tests to supplement b/c I'm not seeing `pk_cpp` generated by test suite for `ceil`, `floor`, `round`, `trunc`..

Reply:
`round` should also work once we add it to the set of allowed math functions